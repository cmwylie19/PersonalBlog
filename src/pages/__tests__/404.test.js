import React from "react"
import * as Gatsby from "gatsby"
import renderer from "react-test-renderer"
import NotFoundPage from "../404"

import { mount } from "enzyme"

describe("404", () => {
  var data
  beforeEach(() => {
    const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery")
    useStaticQuery.mockImplementation(() => ({
      site: {
        siteMetadata: {
          author: "Case",
          description: "Test Page",
          title: "Test Component",
          social: { twitter: "cmwylie19" },
        },
      },
    }))

    data = {
      site: {
        siteMetadata: { social: { twitter: "cmwylie19" }, title: "test" },
      },
    }
  })

  it("renders without crashing", () => {
    mount(<NotFoundPage data={data} location="/" />)
  })
  it("renders correctly", () => {
    const tree = renderer
      .create(<NotFoundPage data={data} location="/" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
