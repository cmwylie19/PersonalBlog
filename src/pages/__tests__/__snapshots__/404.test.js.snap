// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 renders correctly 1`] = `
<div
  style={
    Object {
      "height": "100vh",
      "width": "100% !important",
    }
  }
>
  <div
    style={
      Object {
        "height": "100vh",
        "marginLeft": "auto",
        "marginRight": "auto",
        "maxWidth": "42rem",
        "padding": "2.625rem 1.3125rem",
        "width": "100% !important",
      }
    }
  >
    <header
      style={
        Object {
          "color": "YellowGreen",
        }
      }
    >
      <h2
        style={
          Object {
            "fontFamily": "'Montserrat', sans-serif",
            "marginTop": 0,
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "boxShadow": "none",
              "color": "inherit",
            }
          }
        >
          test
        </a>
      </h2>
    </header>
    <main>
      <h1>
        Not Found
      </h1>
      <p>
        You just hit a route that doesn't exist... the sadness.
      </p>
    </main>
    <footer
      style={
        Object {
          "alignItems": "center",
          "color": "grey",
          "fontFamily": "'Montserrat', sans-serif",
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      Â© 
      2020
      , Built by
       
      <a
        href="mailto:casewylie@gmail.com?subject=blog"
        rel="noopener noreferrer"
        target="_blank"
      >
        Casey
      </a>
      <br />
      <img
        src="https://github.com/cmwylie19/BligBlogger/workflows/Node.js%20CI/badge.svg?event=push"
      />
       
      <img
        src="https://api.netlify.com/api/v1/badges/52f71f5e-6936-4ddf-94e5-049096e6f0f0/deploy-status"
      />
       
    </footer>
  </div>
</div>
`;
